vyper/exceptions.py

```diff
@@ -51,74 +51,117 @@ def __str__(self):
 
 class PythonSyntaxException(ParserException):
     """
-    Conversion from error using ast.parse()
+    Invalid syntax.
+
+    This error is converted from errors raised during ast.parse
     """
     def __init__(self, syntax_error: SyntaxError, source_code: str):
         item = types.SimpleNamespace()  # TODO: Create an actual object for this
         item.lineno = syntax_error.lineno
         item.col_offset = syntax_error.offset
-        item.source_code = source_code
+        item.full_source_code = source_code
         super().__init__(message=f'SyntaxError: {syntax_error.msg}', item=item)
 
 
-class VariableDeclarationException(ParserException):
-    pass
+class SyntaxException(ParserException):
+    """Valid Python syntax, but invalid Vyper syntax."""
 
 
 class StructureException(ParserException):
-    pass
-
-
-class ConstancyViolationException(ParserException):
-    pass
-
-
-class NonPayableViolationException(ParserException):
-    pass
+    """TODO"""
+    # unknown / duplicate interface
+    # invalid map declaration
+    # invalid decorator
+    # use of semi-colon
+    # raise without reason
+    # iteration over nested list
+    # tuple length mismatch
+    # invalid assignment with/without value
+    # improperly formatted struct declaration
+    # invalid global keyword
+    # empty list/tuple
+    # unsupported statement type
+    # altering list during iteration
+    # invalid use of range()
 
 
 class InvalidLiteralException(ParserException):
-    pass
+    """Invalid literal value."""
+    # number out of range
+    # too many decimals
```

Comment:
For numerics that are out of bounds or decimals with too many places.. `OverflowException`?

Reply:
That is much clearer